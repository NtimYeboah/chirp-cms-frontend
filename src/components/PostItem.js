/**
 * Created by ntimobedyeboah on 4/15/17.
 */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import ReactMarkdown from 'react-markdown';

import UserProfile from './UserProfile';

class PostItem extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className="column is-4">
                <Link to={ `posts/${ this.props.details.slug }`}>
                <div className="card">
                    <div className="card-content">
                        <p className="title is-4">{ this.props.details.title }</p>
                        <div className="content post-box">
                            <ReactMarkdown source={ this.props.details.content } />
                        </div>
                        <UserProfile />
                    </div>
                </div>
                </Link>
            </div>
        )
    }
}

export default PostItem;