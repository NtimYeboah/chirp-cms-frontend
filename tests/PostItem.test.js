/**
 * Created by ntimobedyeboah on 4/30/17.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';

import PostItem from './../src/components/PostItem';

describe('PostItem snapshot', () => {
    test('it should render PostItem component and take a snapshot', () => {
        const post = { id: 1234, title: 'Lorem ipsum', content: 'Dolor sit amet', slug: 'lorem-ipsum' };

        let component = renderer.create(
            <MemoryRouter>
                <PostItem details={ post } key={ post.id } />
            </MemoryRouter>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('PostItem rendered output', () => {
    let mountedPostItemComponent;
    const post = { id: 1234, title: 'Lorem ipsum', content: 'Dolor sit amet', slug: 'lorem-ipsum' };


    const postItem = () => {
        if (! mountedPostItemComponent) {
            mountedPostItemComponent = shallow(
                <MemoryRouter>
                    <PostItem details={ post } key={ post.id } />);
                </MemoryRouter>
            );
        }

        return mountedPostItemComponent;
    };

    beforeEach(() => {
        mountedPostItemComponent = undefined;
    });

    test('it should set PostItem props', () => {
        expect(postItem().props().children[0].props.details.title).toBe('Lorem ipsum');
        expect(postItem().props().children[0].props.details.content).toBe('Dolor sit amet');
        expect(postItem().props().children[0].props.details.slug).toBe('lorem-ipsum');
    })
});