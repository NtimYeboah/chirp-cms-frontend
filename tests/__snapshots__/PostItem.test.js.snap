exports[`PostItem snapshot it should render PostItem component and take a snapshot 1`] = `
<div
  className="column is-4">
  <a
    href="posts/lorem-ipsum"
    onClick={[Function]}>
    <div
      className="card">
      <div
        className="card-content">
        <p
          className="title is-4">
          Lorem ipsum
        </p>
        <div
          className="content post-box">
          <div>
            <p>
              Dolor sit amet
            </p>
          </div>
        </div>
        <div
          className="media">
          <div
            className="media-left">
            <figure
              className="image is-48x48">
              <img
                alt=""
                src="http://placehold.it/96x96" />
            </figure>
          </div>
          <div
            className="media-content">
            <p
              className="subtitle is-6">
              John Smith 
              <br />
              <small>
                1 Jan 2016
              </small>
            </p>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
